server/cp/ingest/parser/cp_onclusive.py

```diff
@@ -0,0 +1,63 @@
+import itertools
+from planning.feed_parsers.onclusive import OnclusiveFeedParser
+from typing import List
+from superdesk import get_resource_service
+
+
+def _get_cv_items(_id: str) -> List:
+    cv = get_resource_service("vocabularies").find_one(req=None, _id=_id)
+    return cv["items"]
+
+
+class CPOnclusiveFeedParser(OnclusiveFeedParser):
+    """
+    Superdesk -CP event parser
+
+    Feed Parser which can parse the Onclusive API Events
+    """
+
+    event = []
+
+    def parse(self, content, provider=None):
+        onclusive_cv_items = _get_cv_items("onclusive_ingest_categories")
+        anpa_categories = _get_cv_items("categories")
+        items = super().parse(content, provider)
+
+        for item in items:
+            category = []
+            if item.get("subject"):
+                for subject in item.get("subject"):
+                    if subject["scheme"] == "onclusive_categories":
+                        onclusive_category = self.is_exist(
+                            onclusive_cv_items, subject["qcode"]
+                        )
+                        if onclusive_category:
+                            anpa_category = self.is_exist(
+                                anpa_categories, onclusive_category["cp_category"]
+                            )
+                            if anpa_category:
+                                category.append(
+                                    {
+                                        "name": anpa_category["name"],
+                                        "qcode": anpa_category["qcode"],
+                                        "translations": anpa_category["translations"],
+                                    }
+                                )
+
+            # remove duplicates
+            item["anpa_category"] = [
+                dict(i)
+                for i, _ in itertools.groupby(
+                    sorted(category, key=lambda k: k["qcode"])
+                )
+            ]
+            self.event.append(item)
+        return self.event
+
+    def is_exist(self, cv_items, qcode):
```

Comment:
I would rename this, seems like this would return boolean. something like `find_cv_item` would be better imo